openmc/openmoc_compatible.py

```diff
@@ -0,0 +1,747 @@
+    OPENMC_UNIVERSES[universe_id] = openmc_universe
+
+    # Add the OpenMOC Universe to the global collection of all OpenMOC Universes
+    OPENMOC_UNIVERSES[universe_id] = openmoc_universe
+
+    return openmoc_universe
+
+
+def get_openmc_universe(openmoc_universe):
+    """Return an OpenMC universe corresponding to an OpenMOC universe.
+
+    Parameters
+    ----------
+    openmoc_universe : openmoc.Universe
+        OpenMOC universe
+
+    Returns
+    -------
+    openmc_universe : openmc.universe.Universe
+        Equivalent OpenMC universe
+
+    """
+
+    cv.check_type('openmoc_universe', openmoc_universe, openmoc.Universe)
+
+    universe_id = openmoc_universe.getId()
+
+    # If this Universe was already created, use it
+    if universe_id in OPENMC_UNIVERSES:
+        return OPENMC_UNIVERSES[universe_id]
+
+    # Create an OpenMC Universe to represent this OpenMOC Universe
+    name = openmoc_universe.getName()
+    openmc_universe = openmc.Universe(universe_id, name)
+
+    # Convert all OpenMOC Cells in this Universe to OpenMC Cells
+    for openmoc_cell in openmoc_universe.getCells():
+        openmc_cell = get_openmc_cell(openmoc_cell)
+        openmc_universe.add_cell(openmc_cell)
+
+    # Add the OpenMC Universe to the global collection of all OpenMC Universes
+    OPENMC_UNIVERSES[universe_id] = openmc_universe
+
+    # Add the OpenMOC Universe to the global collection of all OpenMOC Universes
+    OPENMOC_UNIVERSES[universe_id] = openmoc_universe
+
+    return openmc_universe
+
+
+def get_openmoc_lattice(openmc_lattice):
+    """Return an OpenMOC lattice corresponding to an OpenMOC lattice.
+
+    Parameters
+    ----------
+    openmc_lattice : openmc.universe.Lattice
+        OpenMC lattice
+
+    Returns
+    -------
+    openmoc_lattice : openmoc.Lattice
+        Equivalent OpenMOC lattice
+
+    """
+
+    cv.check_type('openmc_lattice', openmc_lattice, openmc.Lattice)
+
+    lattice_id = openmc_lattice.id
+
+    # If this Lattice was already created, use it
+    if lattice_id in OPENMOC_LATTICES:
+        return OPENMOC_LATTICES[lattice_id]
+
+    # Create an OpenMOC Lattice to represent this OpenMC Lattice
+    name = openmc_lattice.name
+    dimension = openmc_lattice.shape
+    pitch = openmc_lattice.pitch
+    lower_left = openmc_lattice.lower_left
+    universes = openmc_lattice.universes
+
+    # Convert 2D dimension to 3D for OpenMOC
+    if len(dimension) == 2:
+        new_dimension = np.ones(3, dtype=np.int)
+        new_dimension[:2] = dimension
+        dimension = new_dimension
+
+    # Convert 2D pitch to 3D for OpenMOC
+    if len(pitch) == 2:
+        new_pitch = np.ones(3, dtype=np.float64) * np.finfo(np.float64).max
+        new_pitch[:2] = pitch
+        pitch = new_pitch
+
+    # Convert 2D lower left to 3D for OpenCG
+    if len(lower_left) == 2:
+        new_lower_left = np.ones(3, dtype=np.float64) * np.finfo(np.float64).min
```

Comment:
Ditto see my message about this above. The joys of binding one's C++ code to a Python front-end.